app/charts/shared/use-sync-interactive-filters.tsx

```diff
@@ -18,15 +18,17 @@ import { useInteractiveFiltersStore } from "@/stores/interactive-filters";
  *
  */
 const useSyncInteractiveFilters = (chartConfig: ChartConfig) => {
-  const {
-    resetCategories,
-    dataFilters,
-    setDataFilters,
-    updateDataFilter,
-    setTimeRange,
-    setCalculationType,
-  } = useInteractiveFiltersStore();
   const { interactiveFiltersConfig } = chartConfig;
+  const resetCategories = useInteractiveFiltersStore((d) => d.resetCategories);
+  const dataFilters = useInteractiveFiltersStore((d) => d.dataFilters);
+  const setDataFilters = useInteractiveFiltersStore((d) => d.setDataFilters);
+  const updateDataFilter = useInteractiveFiltersStore(
+    (d) => d.updateDataFilter
+  );
+  const setTimeRange = useInteractiveFiltersStore((d) => d.setTimeRange);
+  const setCalculationType = useInteractiveFiltersStore(
+    (d) => d.setCalculationType
+  );
```

Comment:
Nit: Wondering, maybe it'd be cool to have next to useInteractiveFIltersStore definition more specialized hooks

const useTimeRange = () => useInteractiveFiltersStore(x => ({ timeRange: x.timeRange, setTimeRange: x.setTimeRange ]) 

etc...

Reply:
Yup, and then we could have the shallow options built-in into this hook, this would be cleaner 👍

Reply:
Will change that in next PR :)